### PR TITLE
Add Release Notes for 4.0

### PR DESCRIPTION
**Pull request by jim-krueger**
Add a section describing the changes made in MP Rest Client 4.0 in the Release Notes section of the specification.

### CODE REVIEW

**Comment by jim-krueger** [comment]
@Emily-Jiang Please review and merge if you agree.   We had some questions as to whether 21 should be listed a a supported Java SE version.

**Review by jamezp — APPROVED** [review]

**Comment by Emily-Jiang** [comment]
There are more things in https://github.com/eclipse/microprofile-rest-client/milestone/10?closed=1. Please list them all. If it is not about API/SPI changes. List them under Other Changes (see guidance [here](https://github.com/eclipse/microprofile/wiki/ReleaseNotes))

File: spec/src/main/asciidoc/release_notes.asciidoc [hunk 1]
```diff
@@ -16,10 +16,43 @@
...
+=== Release Notes for MicroProfile Rest Client 4.0
+
+A full list of changes delivered in the 4.0 release can be found at
+link:https://github.com/eclipse/microprofile-rest-client/milestone/10?closed=1[MicroProfile Rest Client 4.0 Milestone].
+
+==== Incompatible Changes
+This release aligns with Jakarta EE 10, so it won't work with earlier versions of Jakarta or Java EE.
+
+==== API/SPI Changes
+- New overloaded `baseUri(String url)` method so users don’t have
```
[the remark this answers is not in the document]
Comment (jim-krueger) [reply spec/src/main/asciidoc/release_notes.asciidoc]: Fixed.

**Review by jamezp — APPROVED** [review]

File: spec/src/main/asciidoc/release_notes.asciidoc [hunk 2]
```diff
@@ -16,10 +16,43 @@
...
+- Add multi-part processing example to specification.
+- Clarify specification that when a client injected as a CDI bean falls out of scope, the client is closed
+
+==== Other Changes
+- TCK tests added to ensure that priorities specified for client request and response filters are honored by implementations.
+- TCK test added for proper handling of `ResponseExceptionMapper` providers on subresource interface method invocations
+
+==== Jakarta Compatibility
+- EE10
+- EE11 (tolerates)
```
Comment (Emily-Jiang) [inline spec/src/main/asciidoc/release_notes.asciidoc:49]: Have you tested it? If not, I suggest not to mention it as EE 11 has not been released. It is very likely to be released after MP 7.0.

Comment (jim-krueger) [reply spec/src/main/asciidoc/release_notes.asciidoc]: Done thanks

**Review by jamezp — APPROVED** [review]

**Review by Emily-Jiang — APPROVED** [review]

**Pull request merged**